modules/tinymce/src/core/main/ts/api/dom/ScriptLoader.ts

```diff
@@ -222,52 +222,66 @@ class ScriptLoader {
       delete self.scriptLoadedCallbacks[url];
     };
 
-    const load = (urls: string[]) => {
-      self.loading = true;
-      return Promise.allSettled(Arr.map(urls, (url): Promise<void> => {
-        // Script is already loaded then execute script callbacks directly
-        if (self.states[url] === LOADED) {
+    const processResults = (results: Array<PromiseSettledResult<void>>): Promise<void> => {
+      const failures = Arr.filter(results, (result): result is PromiseRejectedResult => result.status === 'rejected');
+      if (failures.length > 0) {
+        return Promise.reject(Arr.bind(failures, ({ reason }) => Type.isArray(reason) ? reason : [ reason ]));
+      } else {
+        return Promise.resolve();
+      }
+    };
+
+    const load = (urls: string[]) => Promise.allSettled(Arr.map(urls, (url): Promise<void> => {
+      // Script is already loaded then execute script callbacks directly
+      if (self.states[url] === LOADED) {
+        execCallbacks('resolve', url);
+        return Promise.resolve();
+      } else if (self.states[url] === FAILED) {
+        execCallbacks('reject', url);
+        return Promise.reject(url);
+      } else {
+        // Script is not already loaded, so load it
+        self.states[url] = LOADING;
+
+        return self.loadScript(url).then(() => {
+          self.states[url] = LOADED;
           execCallbacks('resolve', url);
-          return Promise.resolve();
-        } else if (self.states[url] === FAILED) {
+
+          // Immediately load additional scripts if any were added to the queue while loading this script
+          const queue = self.queue;
+          if (queue.length > 0) {
+            self.queue = [];
+            return load(queue).then(processResults);
```

Comment:
Does this mean that if plugins load additional resources other than translations it will hold up the editor load?

We'll need to audit any plugin that loads non-essential resources to make sure it happens on init rather than blocking editor load.

Reply:
Yes, this is how it worked in v5 as well so it shouldn't make any difference there. Also most plugins don't use the ScriptLoader queue, they just use the `loadScript` API so that wouldn't cause it to block.